src/main/java/com/gooddata/http/client/GoodDataHttpClient.java

```diff
@@ -272,23 +303,20 @@ public HttpResponse execute(HttpUriRequest request, HttpContext context) throws
 
     @Override
     public HttpResponse execute(HttpHost target, HttpRequest request, HttpContext context) throws IOException {
-        if (context == null) {
-            context = this.context;
-        }
-        final ReadWriteLock rwLock = (ReadWriteLock) context.getAttribute(LOCK_RW);
-        Lock readLock = null;
-        if (rwLock != null) {
-            readLock = rwLock.readLock();
-            readLock.lock();
+        final Lock readLock = rwLock.readLock();
+        readLock.lock();
+
+        if (tt != null) {
+            // this adds TT header to EVERY request to ALL hosts made by this HTTP client
```

Comment:
ok, so if the only reason to add TT to EVERY request is webdav, please say it in comment explicitly (hopefully one day it will make easier to clean this up)